hash.c

```diff
@@ -2566,23 +2573,36 @@ rb_hash_update_block_i(VALUE key, VALUE value, VALUE hash)
  *
  *     h1 = { "a" => 100, "b" => 200 }
  *     h2 = { "b" => 254, "c" => 300 }
- *     h1.merge!(h2) { |key, v1, v2| v1 }
- *                     #=> {"a"=>100, "b"=>200, "c"=>300}
- *     h1              #=> {"a"=>100, "b"=>200, "c"=>300}
+ *     h3 = { "b" => 100, "d" => 400 }
+ *     h1.merge!(h2, h3)
+ *                     #=> {"a"=>100, "b"=>100, "c"=>300, "d"=>400}
+ *     h1              #=> {"a"=>100, "b"=>100, "c"=>300, "d"=>400}
+ *
+ *     h1 = { "a" => 100, "b" => 200 }
+ *     h2 = { "b" => 254, "c" => 300 }
+ *     h3 = { "b" => 100, "d" => 400 }
+ *     h1.merge!(h2, h3) { |key, v1, v2| v1 }
+ *                     #=> {"a"=>100, "b"=>200, "c"=>300, "d"=>400}
+ *     h1              #=> {"a"=>100, "b"=>200, "c"=>300, "d"=>400}
  */
 
 static VALUE
-rb_hash_update(VALUE hash1, VALUE hash2)
+rb_hash_update(int argc, VALUE *argv, VALUE self)
 {
-    rb_hash_modify(hash1);
-    hash2 = to_hash(hash2);
-    if (rb_block_given_p()) {
-	rb_hash_foreach(hash2, rb_hash_update_block_i, hash1);
-    }
-    else {
-	rb_hash_foreach(hash2, rb_hash_update_i, hash1);
+    int i;
+    bool block_given = rb_block_given_p();
+
+    rb_hash_modify(self);
+    for (i = 0; i < argc; i++){
+       VALUE hash = to_hash(argv[i]);
+       if (block_given) {
+          rb_hash_foreach(hash, rb_hash_update_block_i, self);
+       }
+       else {
+          rb_hash_foreach(hash, rb_hash_update_i, self);
+       }
```

Comment:
very trivial: These lines are indented with 3 spaces from `for` loop, but maybe usually it's indented with 4 spaces

Reply:
Fixed.  Thank you.